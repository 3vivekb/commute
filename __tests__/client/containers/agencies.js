/* global describe, expect, it */

import React from 'react'
import TestUtils from 'react-addons-test-utils'
import ReactDOM from 'react-dom'
import {Provider} from 'react-redux'

import {makeMockStore, mockStores} from '../../test-utils/mock-data'

import Agencies from '../../../client/containers/agencies'

describe('Container > Agencies', () => {
  it('Agencies View loads', () => {
    // mount component
    const tree = TestUtils.renderIntoDocument(
      <Provider store={makeMockStore(mockStores.init)}>
        <Agencies />
      </Provider>
    )
    const agenciesNode = ReactDOM.findDOMNode(tree)

    console.log(agenciesNode)
  })

  /* it('Agencies View loads with one organization', () => {
    // mount component
    const tree = mount(
      <Provider store={makeMockStore(mockStores.oneSimpleOrganization)}>
        <Agencies />
      </Provider>
    )
    expect(mountToJson(tree.find(Agencies))).toMatchSnapshot()
  })

  it('Delete Organization', () => {
    const mockStore = makeMockStore(mockStores.oneSimpleOrganization)
    window.confirm = () => true

    // Given a logged-in user is viewing the agencies view
    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <Agencies />
      </Provider>
    )

    // When the user clicks the delete button for an existing organization
    // And the user confirms the Confirm Deletion dialog
    const deleteButton = tree.find('table').find(Button).last()
    deleteButton.simulate('click')

    expectDeleteOrganization(mockStore.getActions())
  }) */
})
