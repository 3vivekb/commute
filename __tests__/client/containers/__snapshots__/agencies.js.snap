exports[`Container > Agencies Agencies View loads 1`] = `
"<div class=\"container\">
  <div class=\"row\">
    <div class=\"col-xs-12\">
      <h2>

        <!-- react-text: 5 -->Agencies<!-- /react-text -->
        <button type=\"button\" class=\"pull-right btn btn-default\">
          <a>

            <!-- react-text: 8 -->Create a new agency<!-- /react-text -->
            <i class=\"fa fa-flag fa-fw \"></i>
          </a> 
        </button>
      </h2>
      <p>

        <!-- react-text: 11 -->An agency is a collection of organizations<!-- /react-text -->
        <i class=\"fa fa-shield fa-fw \"></i>

        <!-- react-text: 13 -->.<!-- /react-text -->
      </p>
      <div class=\"react-bs-table-container\">
        <div class=\"react-bs-table\" style=\"height: 100%;\">
          <div class=\"react-bs-container-header table-header-wrapper\">
            <table class=\"table table-hover table-bordered\">
              <thead>
                <tr>
                  <th class=\"\" style=\"text-align: left; display: none;\" data-field=\"id\">
                    <div></div>
                  </th>
                  <th class=\"\" title=\"Name\" style=\"text-align: left;\" data-field=\"name\">

                    <!-- react-text: 23 -->Name<!-- /react-text -->
                    <div></div>
                  </th>
                  <th class=\"\" title=\"Sites\" style=\"text-align: left;\" data-field=\"organizations\">

                    <!-- react-text: 26 -->Sites<!-- /react-text -->
                    <div></div>
                  </th>
                  <th class=\"\" title=\"Tools\" style=\"text-align: left;\" data-field=\"undefined\">

                    <!-- react-text: 29 -->Tools<!-- /react-text -->
                    <div></div>
                  </th>
                </tr>
              </thead>
            </table>
          </div>
          <div class=\"react-bs-container-body\" style=\"height: 100%;\">
            <table class=\"table table-bordered\">
              <colgroup>
                <col class=\"\" style=\"display: none;\">
                  <col class=\"\">
                    <col class=\"\">
                      <col class=\"\">
              </colgroup>
              <tbody>
                <tr class=\"\">
                  <td colspan=\"4\" class=\"react-bs-table-no-data\">There is no data to display</td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`Container > Agencies Agencies View loads with one agency 1`] = `
"<div class=\"container\">
  <div class=\"row\">
    <div class=\"col-xs-12\">
      <h2>

        <!-- react-text: 5 -->Agencies<!-- /react-text -->
        <button type=\"button\" class=\"pull-right btn btn-default\">
          <a>

            <!-- react-text: 8 -->Create a new agency<!-- /react-text -->
            <i class=\"fa fa-flag fa-fw \"></i>
          </a> 
        </button>
      </h2>
      <p>

        <!-- react-text: 11 -->An agency is a collection of organizations<!-- /react-text -->
        <i class=\"fa fa-shield fa-fw \"></i>

        <!-- react-text: 13 -->.<!-- /react-text -->
      </p>
      <div class=\"react-bs-table-container\">
        <div class=\"react-bs-table\" style=\"height: 100%;\">
          <div class=\"react-bs-container-header table-header-wrapper\">
            <table class=\"table table-hover table-bordered\">
              <thead>
                <tr>
                  <th class=\"\" style=\"text-align: left; display: none; min-width: NaNpx;\" data-field=\"id\">
                    <div></div>
                  </th>
                  <th class=\"\" title=\"Name\" style=\"text-align: left; min-width: NaNpx;\" data-field=\"name\">

                    <!-- react-text: 23 -->Name<!-- /react-text -->
                    <div></div>
                  </th>
                  <th class=\"\" title=\"Sites\" style=\"text-align: left; min-width: NaNpx;\" data-field=\"organizations\">

                    <!-- react-text: 26 -->Sites<!-- /react-text -->
                    <div></div>
                  </th>
                  <th class=\"\" title=\"Tools\" style=\"text-align: left; min-width: NaNpx;\" data-field=\"undefined\">

                    <!-- react-text: 29 -->Tools<!-- /react-text -->
                    <div></div>
                  </th>
                </tr>
              </thead>
            </table>
          </div>
          <div class=\"react-bs-container-body\" style=\"height: 100%;\">
            <table class=\"table table-bordered\">
              <colgroup>
                <col class=\"\" style=\"display: none;\">
                  <col class=\"\">
                    <col class=\"\">
                      <col class=\"\">
              </colgroup>
              <tbody>
                <tr class=\"\">
                  <td class=\"\" style=\"text-align: left; display: none;\">agency-1</td>
                  <td class=\"\" style=\"text-align: left;\">
                    <a>Mock Agency</a> 
                  </td>
                  <td class=\"\" style=\"text-align: left;\">
                    <div>0</div>
                  </td>
                  <td class=\"\" style=\"text-align: left;\">
                    <div>
                      <button type=\"button\" class=\"btn btn-warning\">
                        <a>Edit</a> 
                      </button>
                      <button type=\"button\" class=\"btn btn-danger\">Delete</button>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`Container > Agencies Delete Agency 1`] = `
Array [
  Object {
    "payload": "agency-1",
    "type": "delete agency",
  },
  Object {
    "payload": Object {
      "args": Array [
        "/",
      ],
      "method": "push",
    },
    "type": "@@router/CALL_HISTORY_METHOD",
  },
]
`;
